bindings/go/runtime/registry.go

```diff
@@ -0,0 +1,221 @@
+package runtime
+
+import (
+	"fmt"
+	"io"
+	"maps"
+	"reflect"
+	"sync"
+
+	"sigs.k8s.io/yaml"
+)
+
+// Scheme is a dynamic registry for Typed types.
+type Scheme struct {
+	mu sync.RWMutex
+	// allowUnknown allows unknown types to be created.
+	// if the constructors cannot determine a match,
+	// this will trigger the creation of an unstructured.Unstructured with NewScheme instead of failing.
+	allowUnknown bool
+	types        map[Type]any
+}
+
+// NewScheme creates a new registry.
+func NewScheme(opts ...SchemeOption) *Scheme {
+	reg := &Scheme{
+		types: make(map[Type]any),
+	}
+	for _, opt := range opts {
+		opt(reg)
+	}
+	return reg
+}
+
+type SchemeOption func(*Scheme)
+
+// WithAllowUnknown allows unknown types to be created.
+func WithAllowUnknown(allowUnknown bool) SchemeOption {
```

Comment:
FWIW I would just remove the allow since this is a bool field, convention is that `WithAllowUnknown` sets it to true.

Reply:
removed bool